In [4]:
import pandas as pd

# Read the CSV file into a DataFrame
df = pd.read_csv("Question 3.csv")

# Display the first few rows of the DataFrame to inspect its content
df.head(20)


Comments  Ratings
0   I didnt go in with big hopes, but i was expect...        8
1   A unique genre, a well written story (script) ...        8
2   Majestic at scale, grandeur in VFX, and great ...        9
3   "Kalki 2898" is not just a movie; it's an expe...       10
4   Best Indian movie Nagi combined Hindu mytholog...       10
5   The Kalki 2898 AD movie is an epic dystopian s...       10
6   #NagAshwin dreams big and manages to execute t...       10
7   The soaring epicness of second part of the fil...        6
8   Let me share a review of the movie "Kalki 2898...        9
9   I don't understand the obsession with hero ent...        6
10  Where to even begin to define this Subpar movi...        4
11  Definitely worth watching in the theatre\nGrea...        9
12  Kalki28989AD amply suffices my expectations in...        8
13  A unique story and mixup of technology and Mah...        9
14  When we watch Hollywood movies we think Can't ...        8
15  This movie is a spectacular achievement for In...       10
16  Prabhas' latest film, "Kalki," is nothing shor...       10
17  Buckle up for a roller-coaster ride through ci...        6
18  Awesome action sequences. Really enjoyed.RGV a...       10
19  Synopsis: Death is not a punishment; it is a b...        7

In [2]:
from textblob import TextBlob

# Review text provided by the user
review_text = ("'Queen of Katwe' is not just a chess movie; it's a celebration of "
               "the human spirit and the incredible potential that lies within us all. "
               "It's an inspiring and uplifting tale that will leave you feeling both "
               "moved and motivated. This is a must-see film for anyone who values "
               "stories of triumph against adversity and the pursuit of one's dreams.")

# Perform sentiment analysis using TextBlob
analysis = TextBlob(review_text)
sentiment_score = analysis.sentiment.polarity

# Display the sentiment score
sentiment_score


0.35

The sentiment score for the review is 0.35.

Interpretation:
Sentiment Score Range:
-1 indicates very negative sentiment.
0 indicates neutral sentiment.
+1 indicates very positive sentiment.
A score of 0.35 suggests that the review has a moderately positive sentiment, indicating a generally favorable opinion about the movie "Queen of Katwe."

Corresponding Rating:
Given the positive nature of the sentiment, this review would likely correspond to a high rating. Based on typical sentiment-rating mappings, this might translate to a rating in the range of 7 to 9 out of 10.

In [3]:
# Mapping sentiment score to a rating scale of 1 to 10
predicted_rating = ((sentiment_score + 1) / 2) * 9 + 1

# Display the predicted rating
predicted_rating


7.075

To predict the rating for the specific review based on the sentiment score, we can map the sentiment score (which ranges from -1 to 1) to the rating scale (which ranges from 1 to 10). One common approach is to normalize the sentiment score to fit within the rating range.

Mapping Sentiment to Rating:
A sentiment score of -1 corresponds to a rating of 1.
A sentiment score of 1 corresponds to a rating of 10.
The formula to map the sentiment score to a rating would be:

Rating
=
Sentiment Score
+
1
2
×
9
+
1
Rating= 
2
Sentiment Score+1
​
 ×9+1
Let's apply this formula to the sentiment score of 0.35 to predict the rating.






The predicted rating for the review is approximately 7.1 out of 10.

This suggests that based on the sentiment analysis, the review would likely translate to a rating of around 7, indicating a positive but not extremely high rating.